services/createFabricObjects.js

```diff
@@ -0,0 +1,364 @@
+import { fabric } from "fabric";
+import createDiffText from "../utils/createDiffText";
+
+fabric.Textbox.prototype.set({
+  _getNonTransformedDimensions() {
+    return new fabric.Point(this.width, this.height).scalarAdd(this.padding);
+  },
+  _calculateCurrentDimensions() {
+    return fabric.util.transformPoint(
+      this._getTransformedDimensions(),
+      this.getViewportTransform(),
+      true,
+    );
+  },
+});
+
+const renderImg = async (nodeJSON, offsetCoord) => {
+  const position = nodeJSON.property.absoluteBoundingBox;
+
+  if (nodeJSON.type === "ELLIPSE") {
+    const imageObject = async (offsetCoordinates, positions) => {
+      const { x, y, width, height } = positions;
+
+      return new Promise(resolve => {
+        const mask = new fabric.Ellipse({
+          left: x + offsetCoordinates.dx,
+          top: y + offsetCoordinates.dy,
+          angle: nodeJSON.property.arcData.endingAngle,
+          rx: width / 2,
+          ry: height / 2,
+          absolutePositioned: true,
+        });
+
+        fabric.Image.fromURL(nodeJSON.property.imageURL, img => {
+          img.set({
+            left: x + offsetCoordinates.dx,
+            top: y + offsetCoordinates.dy,
+          });
+
+          img.scaleToWidth(width);
+          img.clipPath = mask;
+
+          resolve(img);
+        });
+      });
+    };
+
+    const fabricImageObject = await imageObject(offsetCoord, position);
+
+    return fabricImageObject;
+  }
+
+  if (nodeJSON.type === "RECTANGLE") {
+    const imageObject = async (offsetCoordinates, positions) => {
+      const { x, y, width, height } = positions;
+
+      return new Promise(resolve => {
+        const mask = new fabric.Rect({
+          left: x + offsetCoordinates.dx,
+          top: y + offsetCoordinates.dy,
+          width,
+          height,
+          rx: nodeJSON.property?.cornerRadius,
+          ry: nodeJSON.property?.cornerRadius,
+          absolutePositioned: true,
+        });
+
+        fabric.Image.fromURL(nodeJSON.property.imageURL, img => {
+          img.set({
+            left: x + offsetCoordinates.dx,
+            top: y + offsetCoordinates.dy,
+          });
+
+          img.scaleToWidth(width);
+          img.clipPath = mask;
+
+          resolve(img);
+        });
+      });
+    };
+
+    const fabricImageObject = await imageObject(offsetCoord, offsetCoord);
+
+    return fabricImageObject;
+  }
+};
+
+const renderRect = async (node, offsetCoordinates) => {
+  const isExistImg = node.property.fills[0]?.imageRef ?? null;
```

Comment:
P1: 해당 부분 축약어도 같이 변경해주시면 좋을 것 같습니다!

```suggestion
  const isExistImage = node.property.fills[0]?.imageRef ?? null;
```